static/app/utils/useKeyPress.tsx

```diff
@@ -3,31 +3,33 @@ import {useEffect, useState} from 'react';
 /**
  * Hook to detect when a specific key is being pressed
  */
-function useKeyPress(targetKey: React.KeyboardEvent['key']) {
+const useKeyPress = (targetKey: string, targetRef?: React.RefObject<HTMLElement>) => {
   const [keyPressed, setKeyPressed] = useState(false);
 
   useEffect(() => {
-    function downHandler({key}: KeyboardEvent) {
+    const downHandler = ({key}: KeyboardEvent) => {
       if (key === targetKey) {
         setKeyPressed(true);
       }
-    }
+    };
 
-    function upHandler({key}: KeyboardEvent) {
+    const upHandler = ({key}: KeyboardEvent) => {
       if (key === targetKey) {
         setKeyPressed(false);
       }
-    }
-    window.addEventListener('keydown', downHandler);
-    window.addEventListener('keyup', upHandler);
+    };
+
+    const current = targetRef?.current ?? document.body;
+    current.addEventListener('keydown', downHandler);
+    current.addEventListener('keyup', upHandler);
 
     return () => {
-      window.removeEventListener('keydown', downHandler);
-      window.removeEventListener('keyup', upHandler);
+      current.removeEventListener('keydown', downHandler);
+      current.removeEventListener('keyup', upHandler);
     };
-  }, [targetKey]);
+  }, [targetKey, targetRef]);
```

Comment:
Hmm, this seems wrong.  `targetRef` won't change even if the inner `current` does.  

IE:  If you use this with a ref associated with a conditional element, or an element inside a loop, that may need to change its values between renders, this effect *does not dispose of an re apply the keybinding* because `targetRef` itself won't change, only the `current` value.

I think the fix is to capture `const current = targetRef?.current ?? document.body;` outside of the use effect and add it to the dependency array here instead.  @albertvaka

Reply:
nice good catch, i like it

Reply:
This still won't work, since changing a `ref`'s current value does *not* cause a re-render, thus the useEffect will never trigger and no comparison of the previous targetRef.current value will ever be made

Reply:
In this case you'll actually probably want to `useState` to track the DOM element and changes to it.

Reply:
Another option -> have the useKeyPress optionally return a wrapper object.  This avoids users of `useKeyPress` from having to provide their own ref and handles conditional or changing ref situations elegantly.

```
useKeyPress = (...) => {
  ....
  const Wrapper = useCallback(({children}: PropsWithChildren<{}>) => {
    const ref = useRef(); 
    useLayoutEffect(() => { // runs after mount and ref has been set, I believe 🤞 
      //  ... execute the dom event handling
    }, [ref]); // in this case, you can guarantee that ref.current never changes because the wrapper itself does not use any conditional or loop
    return <React.Fragment ref={ref}>
      {children}
    </>
  }, [setKeyPressed]);

  return [keyPressed, setKeyPressed, Wrapper];
}
```

Reply:
That also won't work, `React.Fragment` doesn't get rendered into the DOM, there will be nothing to attach the event handler to. (Also fragments can't have Refs)

Reply:
I also probably wouldn't recommend arbitrarily wrapping in a `div` or something similar, since then you'll need to deal with it having focus to receive the keyboard events.

I think in this case it would be best to leave deciding how to pass the ref up to the calling component.